config/routes.rb

```diff
@@ -99,6 +99,7 @@
     get 'registrations/psych-sheet' => 'registrations#psych_sheet', as: :psych_sheet
     get 'registrations/psych-sheet/:event_id' => 'registrations#psych_sheet_event', as: :psych_sheet_event
     resources :registrations, only: [:index, :update, :create, :edit, :destroy], shallow: true
+    get 'registrations/:registration_id/payment' => 'payment#registration_payments', as: :registration_payments
```

Comment:
Nit 2: You can probably move this under the `resources :registrations` block with a `do`:
```suggestion
    resources :registrations, only: [:index, :update, :create, :edit, :destroy], shallow: true do
      get 'payments' => 'payment#registration_payments', as: :registration_payments
    end
```

Reply:
Although from the top of my head I wouldn't know how this affects the naming of the route helper. You will have to try that out locally.

Reply:
well, then we should move all those routes there right?